app/services/promo/registrations_stats_fetcher.rb

```diff
@@ -17,17 +18,18 @@ def perform
         request.headers["Content-Type"] = "application/json"
         request.url("/api/2/promo/statsByReferralCode#{query_string}")
       end
-
       referral_code_events_by_date = JSON.parse(response.body)
-
       referral_code_batch.each do |referral_code|
         promo_registration = PromoRegistration.find_by_referral_code(referral_code)
         promo_registration.stats = referral_code_events_by_date.select {|referral_code_event_date|
           referral_code_event_date["referral_code"] == referral_code
         }.to_json
         promo_registration.save!
       end
+      stats = stats + referral_code_events_by_date
     end
+
+    stats
```

Comment:
I made this change so we can directly use the result of this service without looking in the database for the promo registration stats.